developers-api/src/main/java/com/quarksy/api/DevelopersResource.java

```diff
@@ -30,30 +30,46 @@
     public Response create(Developer developer)
     {
         logger.info("POST Developer");
-        return service.addDev(developer);
+        try {
+            return service.addDev(developer);
+        } catch (Exception e) {
+            throw new WebApplicationException(Response.status(BAD_REQUEST).entity("An error has occurred.").build());
+        }
     }
 
     @GET
     public List<Developer> getAllDevs()
     {
         logger.info("Get All Developers");
-        return service.getAllDevs();
+        try {
+            return service.getAllDevs();
+        } catch (Exception e) {
+            throw new WebApplicationException(Response.status(BAD_REQUEST).entity("An error has occurred.").build());
+        }
     }
- 
+
     @GET
     @Path("{id}")
     public Developer getDevById(@PathParam("id") String id)
     {
         logger.info("Get Developer By ID");
-        return service.getDevById(id);
+        try {
+            return service.getDevById(id);
+        } catch (Exception e) {
+            throw new WebApplicationException(Response.status(BAD_REQUEST).entity("An error has occurred.").build());
```

Comment:
I think we should return error message "No developer with that ID was found' or something similar.